resourcegroupstaggingapi_resources - new data source (#17804)
hashicorp/terraform-provider-aws: website/docs/d/resourcegroupstaggingapi_resources.html.markdown

* `resource_type_filter` - (Optional) The constraints on the resources that you want returned. The format of each resource type is `service:resourceType`. For example, specifying a resource type of `ec2` returns all Amazon EC2 resources (which includes EC2 instances). Specifying a resource type of `ec2:instance` returns only EC2 instances.
* `resource_arn_list` - (Optional) Specifies a list of ARNs of resources for which you want to retrieve tag data. Conflicts with `filter`.

### Tag Filter

A `tag_filter` block supports the following arguments:

If you do specify `tag_filter`, the response returns only those resources that are currently associated with the specified tag.
If you don't specify a `tag_filter`, the response includes all resources that were ever associated with tags. Resources that currently don't have associated tags are shown with an empty tag set.

* `key` - (Required) One part of a key-value pair that makes up a tag.
* `values` - (Optional) The optional part of a key-value pair that make up a tag.

## Attributes Reference

In addition to all arguments above, the following attributes are exported:

> * `resource_tag_mapping_list` - A [Resource Tag Mapping List](#resource-tag-mapping-list)`resource_tag_mapping_list` block. documented below.
> 
> ### Resource Tag Mapping List
> 
> A `resource_tag_mapping_list` block supports the following attributes:
> 
> * `resource_arn` - The ARN of the resource.
> * `compliance_details` - Information that shows whether a resource is compliant with the effective tag policy, including details on any noncompliant tag keys. Documented below.
> * `tags` - tags assigned to the resource.

Reviewer: Just to hopefully clarify this attribute a little better:

```suggestion
* `resource_tag_mapping_list` - List of objects matching the search criteria.
    * `compliance_details` - List of objects with information that shows whether a resource is compliant with the effective tag policy, including details on any noncompliant tag keys.
        * `compliance_status` - Whether the resource is compliant.
        * `keys_with_noncompliant_values ` - Set of tag keys with non-compliant tag values.
        * `non_compliant_keys ` - Set of non-compliant tag keys.
    * `resource_arn` - ARN of the resource.
    * `tags` - Map of tags assigned to the resource.
```

---
subcategory: "Resource Groups Tagging API"
layout: "aws"
page_title: "AWS: aws_resourcegroupstaggingapi_resources"
description: |-
  Provides details about resource tagging.
---

# Data Source: aws_resourcegroupstaggingapi_resources

Provides details about resource tagging.

## Example Usage

### Get All Resource Tag Mappings

```terraform
data "aws_resourcegroupstaggingapi_resources" "test" {}
```

### Filter By Tag Key and Value

```terraform
data "aws_resourcegroupstaggingapi_resources" "test" {

>   filter {

Reviewer: ```suggestion
  tag_filter {
```

    key    = "tag-key"
    values = ["tag-value-1", "tag-value-2"]
  }
}
```

### Filter By Resource Type

```terraform
data "aws_resourcegroupstaggingapi_resources" "test" {
  resource_type_filter = ["ec2:instance"]
}
```

## Argument Reference

The following arguments are supported:

* `exclude_compliant_resources` - (Optional) Specifies whether to exclude resources that are compliant with the tag policy. You can use this parameter only if the `include_compliance_details` argument is also set to `true`.
* `include_compliance_details` - (Optional) Specifies whether to include details regarding the compliance with the effective tag policy.

> * `filter` - (Optional) Specifies a list of Tag Filters (keys and values) to restrict the output to only those resources that have the specified tag and, if included, the specified value. See [Filter](#filter) below. Conflicts with `resource_arn_list`.

Reviewer: ```suggestion
* `tag_filter` - (Optional) Specifies a list of Tag Filters (keys and values) to restrict the output to only those resources that have the specified tag and, if included, the specified value. See [Tag Filter](#tag-filter) below. Conflicts with `resource_arn_list`.
```

* `resource_type_filter` - (Optional) The constraints on the resources that you want returned. The format of each resource type is `service:resourceType`. For example, specifying a resource type of `ec2` returns all Amazon EC2 resources (which includes EC2 instances). Specifying a resource type of `ec2:instance` returns only EC2 instances.
* `resource_arn_list` - (Optional) Specifies a list of ARNs of resources for which you want to retrieve tag data. Conflicts with `filter`.

> ### Filter
> 
> A `filter` block supports the following arguments:
> 
> If you do specify `filter`, the response returns only those resources that are currently associated with the specified tag.
> If you don't specify a `filter`, the response includes all resources that were ever associated with tags. Resources that currently don't have associated tags are shown with an empty tag set.

Reviewer: ```suggestion
### Tag Filter

A `tag_filter` block supports the following arguments:

If you do specify `tag_filter`, the response returns only those resources that are currently associated with the specified tag.
If you don't specify a `tag_filter`, the response includes all resources that were ever associated with tags. Resources that currently don't have associated tags are shown with an empty tag set.
```

---
subcategory: "Resource Groups Tagging API"
layout: "aws"
page_title: "AWS: aws_resourcegroupstaggingapi_resources"
description: |-
  Provides details about resource tagging.
---

# Data Source: aws_resourcegroupstaggingapi_resources

Provides details about resource tagging.

## Example Usage

### Get All Resource Tag Mappings

```hcl
data "aws_resourcegroupstaggingapi_resources" "test" {}
```

### Filter By Tag Key and Value

```hcl
data "aws_resourcegroupstaggingapi_resources" "test" {
  tag_filters {
    key    = "tag-key"
    values = ["tag-value-1", "tag-value-2"]
  }
}
```

### Filter By Resource Type

```hcl
data "aws_resourcegroupstaggingapi_resources" "test" {
  resource_type_filter = ["ec2:instance"]
}
```

## Argument Reference

The following arguments are supported:

* `exclude_compliant_resources` - (Optional) Specifies whether to exclude resources that are compliant with the tag policy. You can use this parameter only if the `include_compliance_details` argument is also set to `true`.
* `include_compliance_details` - (Optional) Specifies whether to include details regarding the compliance with the effective tag policy.

> * `tag_filters` - (Optional) Specifies a list of Tag Filters (keys and values) to restrict the output to only those resources that have the specified tag and, if included, the specified value. See [Tag Filters](#tag-filters) below. Conflicts with `resource_arn_list`.

Reviewer 2: I suggest that this attribute is renamed to `tag_filter` to match the `filter` attribute for EC2 (and other service) data sources.

> * `resource_type_filter` - (Optional) The constraints on the resources that you want returned. The format of each resource type is `service:resourceType`. For example, specifying a resource type of ec2 returns all Amazon EC2 resources (which includes EC2 instances). Specifying a resource type of `ec2:instance` returns only EC2 instances.

Reviewer 2: ```suggestion
* `resource_type_filter` - (Optional) The constraints on the resources that you want returned. The format of each resource type is `service:resourceType`. For example, specifying a resource type of `ec2` returns all Amazon EC2 resources (which includes EC2 instances). Specifying a resource type of `ec2:instance` returns only EC2 instances.
```
